mobility_games/games/semantic_waypoints.py

```diff
@@ -0,0 +1,292 @@
+                        AprilTagDetectionArray,
+                        self.tag_callback)
+        rospy.Subscriber('/keyboard/keydown', Key, self.run_mode)
+        srv = Server(SemanticWaypointsConfig, self.config_callback)
+    
+    def config_callback(self, config, level):
+        self.proximity_to_destination = config['proximity_to_destination']
+        return config
+   
+    
+    # FOR DEMO ONLY
+    #   - Creates fake messages
+    def create_test_messages(self):
+       messages = []
+       message_1 = "Hi there! You found it!"
+       messages.append(message_1)
+       self.tag_messages[3] = messages
+
+    def process_pose(self, msg):       #    Onngoing function that updates current x, y, and yaw
+        if self.tag_id:
+            try:
+                self.listener.waitForTransform('odom', 
+                                               msg.header.frame_id, 
+                                               msg.header.stamp, 
+                                               rospy.Duration(0.5))
+                newitem = self.listener.transformPose('odom', msg)
+                self.x = newitem.pose.position.x
+                self.y = newitem.pose.position.y
+                
+                if self.translations:
+                    self.distance_to_destination = math.sqrt((self.translations[0] - self.x)**2 + (self.translations[1] - self.y)**2)
+                angles = euler_from_quaternion([msg.pose.orientation.x,
+                                                msg.pose.orientation.y,
+                                                msg.pose.orientation.z,
+                                                msg.pose.orientation.w])
+                self.yaw = angles[2]
+
+            except Exception as inst:
+                print "Exception is", inst
+
+    def det_speech(self, yaw, cur_pos, goal_pos):
+        print type(cur_pos[0])
+
+        #   Takes in angle, position, and position of goal. Determines relative
+        #   position of goal to Tango and returns text instructions.
+        dx = goal_pos[0] - cur_pos[0]
+        dy = goal_pos[1] - cur_pos[1]
+        angle = math.atan2(dy, dx) - yaw
+        angle %= 2*math.pi      #   Makes sure angle is between 0 and 2pi
+        if angle < math.pi/6:
+            text = "You're on the right path!"
+        elif angle < math.pi/3:
+            text = "The goal is slightly to your left."
+        elif angle < 2*math.pi/3:
+            text = "The goal is to your left."
+        elif angle < 4*math.pi/3:
+            text = "The goal is behind you."
+        elif angle < 5*math.pi/3:
+            text = "The goal is to your right."
+        elif angle < 2*math.pi:
+            text = "The goal is slightly right."
+        else:
+            text = "Something funny happened."
+        return text
+
+    # Adds string labels to AR Tags
+    def calibrate_tag(self, tag_id):    
+        self.visited_tags_calibrate.append(tag_id); 
```

Comment:
`        self.visited_tags_calibrate.append(tag_id); `
`tag_name = raw_input("Name the AR Tag: "); `

Do semicolons do anything in Python? Probably should clean them up for consistency.